src/main/java/com/company/library/CollectionTask.java

```diff
@@ -32,69 +39,119 @@ public static void main(String[] args) throws IOException, URISyntaxException {
                     authorBooks.size(),
                     authorBooks);
 
-        }
+        }*/
+
+        booksByAuthor.entrySet().stream().forEach(entry -> {
+            final var author = entry.getKey();
+            final var authorBooks = entry.getValue();
+            System.out.printf("Author %s %s has %d books in library: %s %n",
+                    author.getFirstName(),
+                    author.getLastName(),
+                    authorBooks.size(),
+                    authorBooks);
+
+        });
 
     }
 
     private static Map<Author, List<Book>> groupBooksByAuthors(List<Book> books) {
-        Map<Author, List<Book>> listMap = new HashMap<>();
+        /*Map<Author, List<Book>> listMap = new HashMap<>();
         for (Book book : books) {
             if (!listMap.containsKey(book.author)) {
                 listMap.put(book.author, new ArrayList<>());
             }
             listMap.get(book.author).add(book);
 
-        }
-
-        return listMap;
+        }*/
+        return books.stream()
+                .collect(Collectors.groupingBy(book -> book.author));
     }
 
     //TODO change return type to appropriate one
     private static Collection<Author> findUniqueAuthors(List<Book> books) {
         //TODO return only unique authors
-        Set<Author> uniqueAuthors = new HashSet<>();
+        /*Set<Author> uniqueAuthors = new HashSet<>();
         for (Book book : books) {
             uniqueAuthors.add(book.author);
-        }
-
-        return uniqueAuthors;
-
+        }*/
+        return books.stream()
+                .map(book -> book.author)
+                .collect(Collectors.toSet());
     }
 
     //TODO change return type to appropriate one
     private static Collection<Book> findUniqueBooks(List<Book> books) {
         //TODO return only unique books
-        return new HashSet<>(books);
+        //return new HashSet<>(books);
+
+        return books.stream().distinct().collect(Collectors.toList());
+
     }
 
     public static List<String> getLinesFromFile(final String filepath) throws IOException, URISyntaxException {
-        return Files.lines(
-                Path.of(
-                        CollectionTask.class
-                                .getClassLoader()
-                                .getResource(filepath)
-                                .toURI()
-                )
-        )
-                .toList();
+
+        try {
+            return Files.lines(
+                    Path.of(
+                            CollectionTask.class
+                                    .getClassLoader()
+                                    .getResource(filepath)
+                                    .toURI()
+                    )
+            )
+                    .toList();
+        } catch (NullPointerException e) {
+            List<String> emptyFile = new ArrayList<>();
+            System.err.println("File doesn't found");
+            return emptyFile;
```

Comment:
You can use the static method `emptyList()` instead of instantiating a new variable;